### PR TITLE
feat(clipboard): add clipboard copy to some helper text fields

### PR DESCRIPTION
**Pull request by maxcao13**
Signed-off-by: Max Cao <macao@redhat.com>

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] Signed the last commit: `git commit --amend --signoff`
_______________________________________________

Fixes: https://github.com/cryostatio/cryostat-web/issues/695

## Description of the change:
This change adds allows a match expression example to be copied to the clipboard. As well as the Add Target modal example.

## Motivation for the change:
This change is helpful because users may want to copy the example for easier use.

## How to manually test:
1. Checkout pr and run the headless image with CORS.
2. Press add target button on dashboard and notice new change and try it.
3. Go to automated rules create form, and notice new change and try it.

### CODE REVIEW

**Comment by maxcao13** [comment]
Create Rule test snapshot doesn;t work because of issues with Popper again. I would think that just mocking away the Popper would work, but that doesn't seem to be the case.

File: src/app/TargetSelect/CreateTargetModal.tsx [hunk 1]
```diff
@@ -135,7 +137,15 @@ export const CreateTargetModal: React.FunctionComponent<CreateTargetModalProps>
             label="Connection URL"
             isRequired
             fieldId="connect-url"
-            helperText="JMX Service URL, e.g. service:jmx:rmi:///jndi/rmi://localhost:0/jmxrmi"
+            helperText={
+              <FormHelperText isHidden={false} component="div">
+                JMX Service URL e.g.
```
Comment (tthvo) [inline src/app/TargetSelect/CreateTargetModal.tsx:142]: Maybe `JMX Service URL. For example:`?

File: src/app/Shared/MatchExpressionEvaluator.tsx [hunk 2]
```diff
@@ -149,7 +149,19 @@ export const MatchExpressionEvaluator: React.FunctionComponent<MatchExpressionEv
     }
     body = JSON.stringify(body, null, 2);
     body = body.substring(1, body.length - 1);
-    return body;
+    return !props.inlineHint ? (
```
Comment (tthvo) [inline src/app/Shared/MatchExpressionEvaluator.tsx:152]: Maybe here can just continue to return the string. And use code block action here:

https://github.com/cryostatio/cryostat-web/blob/f8ae71e528a2ada04da5975e49efc223dbcceaff/src/app/Shared/MatchExpressionEvaluator.tsx#L186

Reference: https://www.patternfly.org/v4/components/code-block

Example:

Screenshot from 2022-12-08 21-45-29

**Comment by tthvo** [comment]
> Create Rule test snapshot doesn;t work because of issues with Popper again (part of ClipboardCopy Tooltip). I would think that just mocking away the Popper would work, but that doesn't seem to be the case.

Looks like we might have to drop it then :(( Tried mocking `Tooltip` itself but does not work (even in `test-setup.js`).

File: src/app/Shared/MatchExpressionEvaluator.tsx [hunk 3]
```diff
@@ -104,6 +107,15 @@ export const MatchExpressionEvaluator: React.FunctionComponent<MatchExpressionEv
     }
   }, [props.onChange, valid]);
 
+  const clipboardCopyFunc = (event, text) => {
```
Comment (tthvo) [inline src/app/Shared/MatchExpressionEvaluator.tsx:110]: React.useCallback for memo?

Comment (tthvo) [reply src/app/Shared/MatchExpressionEvaluator.tsx:110]: Same for `onClick` too?

Comment (tthvo) [reply src/app/Shared/MatchExpressionEvaluator.tsx:110]: Maybe we can just group them all in a single function:

```typescript
const onSaveToClipBoard = React.useCallback(
  (_) => {
    setCopied(true);
    navigator.clipboard.writeText(exampleExpression);
  },
  [navigator.clipboard, exampleExpression, setCopied]
);
```

File: src/app/Shared/MatchExpressionEvaluator.tsx [hunk 4]
```diff
@@ -152,6 +164,27 @@ export const MatchExpressionEvaluator: React.FunctionComponent<MatchExpressionEv
...
+            id="match-expression-copy-button"
+            textId="match-expression-code-content"
+            aria-label="Copy to clipboard"
+            onClick={(e) => onClick(e, exampleExpression)}
+            exitDelay={copied ? 1500 : 600}
+            maxWidth="110px"
+            variant="plain"
+            onTooltipHidden={() => setCopied(false)}
+          >
+            {copied ? 'Successfully copied to clipboard!' : 'Copy to clipboard'}
```
Comment (tthvo) [inline src/app/Shared/MatchExpressionEvaluator.tsx:181]: Might need to sync this with CreateTargetModal clipboard message:

```typescript
{copied ? 'Copied!' : 'Click to copy to clipboard'}
```

**Review by tthvo — APPROVED** [review]
Looks good to me :D

**Pull request merged**